python/TestHarness/util.py

```diff
@@ -414,6 +414,44 @@ def getSortedValuesSets(self):
             d = dict(((k, v-t) for k, v in d.items() if v))
         return r
 
+# Calculates the Reverse Reachability of a graph. This class does this by
+# computing the reachability of a graph (reversing the direction of edges)
+class ReverseReachability:
+    def __init__(self):
+        self.dependency_dict = {}
+
+    def insertDependency(self, key, values):
+        # Insert the dependencies in backwards
+        for value in values:
+            self.dependency_dict.setdefault(value, set(key)).add(key)
```

Comment:
@milljm found that the tests didn't work when the keys are multiple characters and all the tests use single characters...